pkg/multicluster/access/kubeconfig_secret.go

```diff
@@ -0,0 +1,97 @@
+package access
+
+import (
+	kubev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+// KubeConfigSecretLabel is applied to a kubernetes secret to indicate that the secret contains a kubeconfig.
+const KubeConfigSecretLabel = "solo.io/kubeconfig"
+
+// KubeConfig wraps kubernetes KubeConfigs and also indicates the name of the cluster the KubeConfig belongs to.
+// Used to persist KubeConfigs to kubernetes secrets.
+type KubeConfig struct {
+	// the actual KubeConfig
+	Config api.Config
+	// expected to be used as an identifier string for a cluster
+	// stored as the key for the KubeConfig data in a kubernetes secret
+	Cluster string
+}
+
+// KubeConfigToSecret converts a single KubeConfig to a secret with the provided name and namespace.
+func KubeConfigToSecret(name string, namespace string, kc *KubeConfig) (*kubev1.Secret, error) {
```

Comment:
are both methods necessary? i wonder if we could live with just one

Reply:
removed the function that took multiple kubeconfigs for now, happy to restore it if we decide to support multiple kubeconfigs per secret.

Reply:
service mesh hub team is moving toward one per secret, seems good to me.